aria_data_loaders/aria_data_utils/aria_live_streamer.py

```diff
@@ -0,0 +1,469 @@
+        self._dst_calib_params = get_linear_camera_calibration(
+            512, 512, 280, "camera-rgb"
+        )
+        self.device_T_camera = sp.SE3(
+            self._sensors_calib.get_transform_device_sensor("camera-rgb").to_matrix()
+        )  # type: ignore
+
+        # Configure subscription to listen to Aria's RGB stream.
+        # streaming configuration
+        config = streaming_client.subscription_config
+        options = aria.StreamingSecurityOptions()
+        options.use_ephemeral_certs = False
+        options.local_certs_root_path = os.path.expanduser(
+            ("~/.aria/streaming-certs/persistent/")
+        )
+        config.security_options = options
+        config.subscriber_data_type = aria.StreamingDataType.Rgb
+        # A shorter queue size may be useful if the processing callback is always slow and you wish to process more recent data
+        # For visualizing the images, we only need the most recent frame so set the queue size to 1
+        # @TODO: Identify the best queue size for your application
+        config.message_queue_size[aria.StreamingDataType.Rgb] = 1  # type: ignore
+        config.message_queue_size[aria.StreamingDataType.Slam] = 1  # type: ignore
+        streaming_client.subscription_config = config
+
+        streaming_client.set_streaming_client_observer(self)
+
+        return device, device_client
+
+    def _rotate_img(self, img: np.ndarray, num_of_rotation: int = 3) -> np.ndarray:
+        """
+        Rotate image in multiples of 90d degrees
+
+        Args:
+            img (np.ndarray): Image to be rotated
+            k (int, optional): Number of times to rotate by 90 degrees. Defaults to 3.
+
+        Returns:
+            np.ndarray: Rotated image
+        """
+        img = np.ascontiguousarray(
+            np.rot90(img, k=num_of_rotation)
+        )  # GOD KNOW WHY THIS IS NEEDED -> https://github.com/clovaai/CRAFT-pytorch/issues/84#issuecomment-574683857
+        return img
+
+    def on_image_received(self, image: np.ndarray, record: ImageDataRecord):
+        if self._is_connected:
+            rospy.logdebug("Received image")
+            self.aria_rgb_frame = image
+
+    def on_pose_received(self, msg: PoseStamped):
+        if self._is_connected:
+            rospy.logdebug("Received pose")
+            self.aria_pose_device = ros_pose_to_sophus(msg.pose)
+            self.aria_pose = self.aria_pose_device * self._device_T_cpf
+            self.aria_ros_pose = sophus_to_ros_pose(self.aria_pose)
+            odom_msg = Odometry()
+            odom_msg.header.frame_id = "ariaWorld"
+            odom_msg.child_frame_id = "ariaDevice"
+            odom_msg.header.stamp = rospy.Time.now()
+            odom_msg.pose.pose = self.aria_ros_pose
+            self.aria_odom_publisher.publish(odom_msg)
+            msg.pose = self.aria_ros_pose
+            self.aria_cpf_publisher.publish(msg)
```

Comment:
Good practice to add the `if` condition check here. But just curious, will we receive any new poses if device got disconnected for whatever reason?